presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestGroupInnerJoinsByConnector.java

```diff
@@ -0,0 +1,531 @@
+
+        CatalogManager catalogManager = new CatalogManager();
+        catalogManager.registerCatalog(new Catalog(
+                catalogName,
+                connectorId,
+                testConnector,
+                createInformationSchemaConnectorId(connectorId),
+                testConnector,
+                createSystemTablesConnectorId(connectorId),
+                testConnector));
+
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Metadata metadata = createTestMetadataManager(transactionManager, new FeaturesConfig(), new FunctionsConfig());
+
+        TransactionId transactionId = transactionManager.beginTransaction(false);
+        Session session = testSessionBuilder().setTransactionId(transactionId).build();
+
+        ImmutableMap.Builder<ColumnHandle, Domain> domains = new ImmutableMap.Builder<>();
+        domains.put(new InformationSchemaColumnHandle("table_schema"), Domain.singleValue(VARCHAR, Slices.utf8Slice("test_schema")));
+        Constraint<ColumnHandle> constraint = new Constraint<>(TupleDomain.withColumnDomains(domains.build()));
+
+        InformationSchemaMetadata informationSchemaMetadata = new InformationSchemaMetadata(catalogName, metadata);
+        informationSchemaMetadata.getTableLayouts(
+                createNewSession(transactionId),
+                new InformationSchemaTableHandle(catalogName, "information_schema", "views"),
+                constraint,
+                Optional.empty());
+
+        VariableReferenceExpression left = newBigintVariable("a1");
+        VariableReferenceExpression right = newBigintVariable("a2");
+        EquiJoinClause joinClause = new EquiJoinClause(left, right);
+
+        PlanNode plan = output(join(FULL,
+                tableScan(connectorName, "a1", "b1"),
+                tableScan(connectorName, "a2", "b2"),
+                joinClause));
+        PlanNode actual = optimize(plan, session, ImmutableMap.of(), metadata);
+        Assert.assertEquals(plan, actual);
+    }
+
+    @Test
+    public void testDoesNotJoinPushDownTwoDifferentConnectors()
+    {
+        String catalog1 = "test_catalog_1";
+        String catalog2 = "test_catalog_2";
+
+        ConnectorId connectorId1 = new ConnectorId(catalog1);
+        ConnectorId connectorId2 = new ConnectorId(catalog2);
+
+        MockConnectorFactory.Builder builder = MockConnectorFactory.builder();
+        MockConnectorFactory mockConnectorFactory = builder.withListSchemaNames(connectorSession -> ImmutableList.of("test_schema"))
+                .withListTables((connectorSession, schemaNameOrNull) ->
+                        ImmutableList.of(
+                                new SchemaTableName("test_schema", "test_view"),
+                                new SchemaTableName("test_schema", "another_table")))
+                .withConnectorCapabilities(ImmutableSet.of()).build();
```

Comment:
Let's have all the connectors SUPPORTS_JOIN_PUSHDOWN. The test should work with or without the connectors supporting it